scripts/deploy-federated-nginx.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script has an end to end steps in deploying a federated deployment
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "$(dirname "${BASH_SOURCE}")/util.sh"
+
+TEST_NS="test-namespace"
+
+# Prerequisite:
+# - a cluster scoped federation deployment with `cluster1` and
+#   `cluster2` joined as member clusters
+# - kubectl config use-context cluster1
+#
+echo "Starting deploying a ngnix application in federation cluster."
+
+echo
+echo "Creating a test namespace ${TEST_NS}."
+kubectl create ns ${TEST_NS}
+
+echo
+echo "Creating a ngnix deployment in namespace ${TEST_NS}."
+kubectl apply -n ${TEST_NS} -f ../example/sample1/configmap.yaml
+
+kubectl apply -n ${TEST_NS} -f ../example/sample1/deployment.yaml
+
+echo
+echo "Creating a ngnix service in namespace ${TEST_NS}."
+kubectl apply -n ${TEST_NS} -f ../example/sample1/service.yaml
+
+echo
+echo "Federating resources in namespace ${TEST_NS} to member clusters."
+kubefedctl federate ns ${TEST_NS} --contents --skip-api-resources 'pods,secrets,serviceaccount,replicaset'
+
+echo
+echo "Checking status of federated resources."
+# TODO: Verify
+
+echo
+echo "Querying status of deployment nginx applications from member clusters."
+IP1=$(minikube -p cluster1 ip)
```

Comment:
This is going to run in CI, and it won't be an option to use multiple minikube clusters.  You'll need to figure out how to do the same thing with kind.